### PR TITLE
[Ready to merge ] Starknet getBlock*

### PR DESCRIPTION
**Pull request by erluxman**

### CODE REVIEW

File: lib/src/provider/read_provider.dart [hunk 1]
```diff
@@ -1,5 +1,9 @@
+import 'package:starknet/src/provider/model/get_block_txn_count.dart';
```
Comment (gabsn) [inline lib/src/provider/read_provider.dart]: never import from deep nested path. You should add your file to the `exports` in the `index.dart`

File: lib/src/provider/read_provider.dart [hunk 2]
```diff
@@ -1,5 +1,9 @@
...
 import 'package:starknet/starknet.dart';
 
+import 'model/get_block_txn_count.dart';
+import 'model/get_block_with_txs.dart';
```
Comment (gabsn) [inline lib/src/provider/read_provider.dart]: don't use relative import, everything should be accessible from "package:starknet/starknet.dart"

File: test/provider/read_provider_test.dart [hunk 3]
```diff
@@ -1,3 +1,5 @@
+import 'package:starknet/src/provider/model/get_block_txn_count.dart';
+import 'package:starknet/src/provider/model/get_block_with_txs.dart';
```
Comment (gabsn) [inline test/provider/read_provider_test.dart]: don't use nested imports

File: test/provider/read_provider_test.dart [hunk 4]
```diff
@@ -300,6 +303,40 @@ void main() {
...
+      test('returns unimplemented method error for getStateUpdate', () async {
+        final GetBlockWithTxs response = await provider.getBlockWithTxs(
+            BlockId.blockHash(
+                blockHash: Felt.fromHexString(
+                    '0x3fbf1b9a9ed822423e87365923103a9577ebed2612afccf4c9f69c126eeeeb7')));
+        response.when(
+            error: (error) {
+              expect(error.code, equals(-32601));
+              expect(error.message,
+                  contains('method \'starknet_getBlockWithTxs\' not found'));
```
Comment (gabsn) [inline test/provider/read_provider_test.dart]: you should be able to use the v010 pathfinder provider for this test now and expect a real value

**Review by gabsn — CHANGES_REQUESTED** [review]
Hey Lux, good work, however a few points to fix before the merge:
- don't use relative or nested imports. Export your models from the `index.dart` instead
- don't use the `infuraGoerliRpc` endpoint. Instead use the `v010PathfinderGoerliTestnetUri` instead. It should have the `getBlockXXX` methods implemented.

**Comment by erluxman** [comment]
The tests are failing coz the server isn't available, please have a look at the tests.

**Comment by gabsn** [comment]
@erluxman do you need help to have the tests pass?

**Review by gabsn — APPROVED** [review]
LGTM

**Pull request merged**